notifications_utils/template.py

```diff
@@ -166,12 +166,12 @@ def prefix(self, value):
 
     @property
     def content_count(self):
```

Comment:
So just to spell out my understanding, this PR is changing `content_count` from instead of returning the number of bytes to return the number of characters. Originally, we've always wanted to return the number of characters and were assuming that the number of bytes was equivalent (although we were wrong). 

To make it clear what this method is purposed for, I'd like to see either the name of the method change to `content_character_count` (however realise this is more work and would mean changing things in other apps) or at the least add a doc string to explicitly say that we return the number of characters please.

Reply:
yeah, your assumption is correct - bytes are irrelevant, and we made a bad assumption four years ago. in the ideal world, bytes should never have come into this, cos at no point have we or the providers ever encoded the unicode string into bytes.

It appears that the providers don't care between us sending `'Â' * 612` and sending `'a' * 612`. It's only when we're calculating billing information that we care about what characters are included, (the first is 10 fragments and the second is only 4) ie in [get_sms_fragment_count](https://github.com/alphagov/notifications-utils/blob/master/notifications_utils/template.py#L628-L632)

i won't change the name partly because it's a relatively unnecessary breaking change, but also it's unneeded clarification as there's no other alternative way to count the content. I can add a docstring though.

Reply:
semi-relatedly, we are using the content_count method elsewhere, but it does appear that we're checking `content_count > SMS_CHAR_COUNT_LIMIT` in three separate places in api/admin, all of which could be replaced with calls to `template.is_message_too_long`. (We still do call the property in a couple of other places)